packages/html-reporter/src/reportView.tsx

```diff
@@ -42,51 +43,75 @@ const testCaseRoutePredicate = (params: URLSearchParams) => params.has('testId')
 export const ReportView: React.FC<{
   report: LoadedReport | undefined,
 }> = ({ report }) => {
-  const searchParams = new URLSearchParams(window.location.hash.slice(1));
+  const searchParams = useSearchParams();
   const [expandedFiles, setExpandedFiles] = React.useState<Map<string, boolean>>(new Map());
   const [filterText, setFilterText] = React.useState(searchParams.get('q') || '');
 
+  const reportData = report?.json();
   const filter = React.useMemo(() => Filter.parse(filterText), [filterText]);
-  const filteredStats = React.useMemo(() => computeStats(report?.json().files || [], filter), [report, filter]);
+
+  const filteredFiles = React.useMemo(() => {
+    const files = reportData?.files ?? [];
+
+    return files
+        .map(file => {
+          const tests = file.tests.filter(test => filter.matches(test));
+          return { ...file, tests };
+        })
+        .filter(file => file.tests.length !== 0);
+  }, [filter, reportData?.files]);
+
+  const filteredStats = React.useMemo(() => computeStats(filteredFiles), [filteredFiles]);
 
   return <div className='htmlreport vbox px-4 pb-4'>
     <main>
-      {report?.json() && <HeaderView stats={report.json().stats} filterText={filterText} setFilterText={setFilterText}></HeaderView>}
-      {report?.json().metadata && <MetadataView {...report?.json().metadata as Metainfo} />}
+      {reportData && <HeaderView stats={reportData.stats} filterText={filterText} setFilterText={setFilterText}></HeaderView>}
```

Comment:
Let's try to avoid stray changes where possible

Reply:
changed